app/src/main/java/com/damianhinch/sunshine/MainActivity.java

```diff
@@ -216,15 +205,37 @@ public boolean onOptionsItemSelected(MenuItem item) {
             startActivity(intentToOpenSettingsActivity);
         }
         if (id == R.id.refresh_button) {
-            final String userLocationPreference = getUserPreferredLocation();
-            final String userUnitsPreference = getUserUnitsPreference();
+            final String userLocationPreference = Helpers.getUserPreferredLocation(this);
```

Comment:
You could put this and the following two lines in a method. Since the same sequence is called in `onResume()`.